Pinta.Effects/Effects/FeatherEffect.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading.Tasks;
+using Cairo;
+using Pinta.Core;
+using Pinta.Gui.Widgets;
+
+namespace Pinta.Effects;
+
+public sealed class FeatherEffect : BaseEffect
+{
+
+	public override string Icon => Pinta.Resources.Icons.EffectsDefault;
+
+	// Multithread internally within FeatherEffect to get the full-sized rois
+	public sealed override bool IsTileable => false;
+
+	public override string Name => Translations.GetString ("Feather");
+
+	public override bool IsConfigurable => true;
+
+	public override string EffectMenuCategory => Translations.GetString ("Stylize");
+
+	public FeatherData Data => (FeatherData) EffectData!;  // NRT - Set in constructor
+
+	private readonly IChromeService chrome;
+	private readonly ISystemService system;
+
+	public FeatherEffect (IServiceProvider services)
+	{
+		chrome = services.GetService<IChromeService> ();
+		system = services.GetService<ISystemService> ();
+		EffectData = new FeatherData ();
+	}
+
+	public override void LaunchConfiguration ()
+		=> chrome.LaunchSimpleEffectDialog (this);
+
+	protected override void Render (ImageSurface src, ImageSurface dest, RectangleI roi)
+	{
+		int bottom = roi.Bottom;
+		int src_height = src.Height;
+		int radius = Data.Radius;
+		int threads = system.RenderThreads;
+
+		ConcurrentBag<PointI> borderPixels = new ConcurrentBag<PointI> ();
+		// Color in any pixel that the stencil says we need to fill
+		// First pass
+		// Clean up dest, then collect all border pixels
+		Parallel.For (0, bottom, new ParallelOptions { MaxDegreeOfParallelism = threads }, y => {
```

Comment:
Should this loop start from the top of the rectangle, rather than the top of the image?
Same for the second parallel loop too

Reply:
Oops... Good catch!